virtual/types.go

```diff
@@ -71,6 +72,14 @@ type Environment interface {
 	// in the registry, but allows us to test interaction between the client versionstamp
 	// and the serverion heartbeat versionstamp.
 	freezeHeartbeatState()
+
+	// prevents the heartbeat goroutine from sending the heartbeat
```

Comment:
super nit: convention for go method comments is something like:

`pauseHeartbeat...` (I.E first word should be method name, linters will complain about this for public methods)

Reply:
also `.` at end